from django.core.management.base import BaseCommand, CommandError
from core.settings import BASE_DIR
import pandas as pd
from Main.models import MD5MalwareHashes
import os
class Command(BaseCommand):
    def handle(self, *args, **kwargs):
        filepath=str(BASE_DIR) + '/MalwareModel/MD5MalwareHashes.csv' 
        if not os.path.exists(filepath):
            self.stdout.write(self.style.ERROR(f"File not found: {filepath}"))
            return
        df = pd.read_csv(filepath)
        for index,row in df.iterrows():
            md5_value = row['MD5']
            try:
                MD5MalwareHashes.objects.create(MD5=md5_value)
                self.stdout.write(self.style.SUCCESS(f"{index} = {md5_value} added to the database"))
            except Exception as e:
                self.stdout.write(self.style.ERROR(f"Error adding {md5_value}: {e}"))